### PR TITLE
Fix Distribution TODOs

### PR DESCRIPTION
**Pull request by a-veitch**
Resolves some TODO's in the original Distribution code:
* Allows for zero-length bucketBoundaries (creates single bucket for over and underflow bucket, matching documentation allowing N=1 total buckets)
* Simplifies Range by not using null for initial min/max.

### CODE REVIEW

File: core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java [hunk 1]
```diff
@@ -95,6 +96,16 @@ long getCount() {
   }
 
   /**
+   * The sum of the squares of values in the population. If {@link #getCount()} is zero then this
+   * value will also be zero.
+   *
+   * @return The sum of squares of values in the population.
+   */
+  double getSumSquares() {
```
Comment (songy23) [inline core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java]: Remove this?

Comment (a-veitch) [inline core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java]: D'oh. Thanks, and done.

File: core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 final class MutableDistribution {
   private long count = 0; // The number of values in the population.
   private double sum = 0.0; // The sum of the values in the population.
+  private double sumOfSquares = 0.0; // The sum of the sqaures of the values in the population
```
Comment (songy23) [inline core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java]: Also remove this? Didn't see we use it anywhere

File: core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java [hunk 3]
```diff
@@ -157,6 +158,7 @@ private void putIntoBucket(double value) {
   void add(double value) {
     count++;
     sum += value;
+    sumOfSquares += value * value;
```
Comment (songy23) [inline core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java]: Same here

File: core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java (hunk 2 above)
Comment (a-veitch) [inline core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java]: double and triple d'oh. Done

File: core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java (hunk 3 above)
Comment (a-veitch) [inline core/src/main/java/com/google/instrumentation/stats/MutableDistribution.java]: and done...

**Pull request merged**